primitives/inherents/src/lib.rs

```diff
@@ -436,31 +436,10 @@ mod tests {
 		assert!(data.put_data(TEST_INHERENT_0, &10).is_err());
 	}
 
-	#[derive(Clone)]
-	struct TestInherentDataProvider;
-
-	const ERROR_TO_STRING: &str = "Found error!";
-
-	#[async_trait::async_trait]
-	impl InherentDataProvider for TestInherentDataProvider {
-		fn provide_inherent_data(&self, data: &mut InherentData) -> Result<(), Error> {
-			data.put_data(TEST_INHERENT_0, &42)
-		}
-
-		async fn try_handle_error(
-			&self,
-			_: &InherentIdentifier,
-			_: &[u8],
-		) -> Option<Result<(), Error>> {
-			Some(Err(Error::Application(Box::from(ERROR_TO_STRING))))
-		}
-	}
-
 	#[test]
 	fn create_inherent_data() {
-		let provider = TestInherentDataProvider;
-
-		let inherent_data = provider.create_inherent_data().unwrap();
+		let provider = substrate_test_runtime_client::runtime::TestInherentDataProvider;
+		let inherent_data = futures::executor::block_on(provider.create_inherent_data()).unwrap();
```

Comment:
Your problem here is that `test-runtime` imports `sp-inherents` and then `sp-inherents` imports `test-runtime`. For rustc the trait `InherentDataProvider` that you get through `test-runtime` is different than the one defined in the crate here. It can not "proof" that both are actually the same. Don't ask me now how this is called :P 

However, you should revert the changes and not define this type in the test runtime.

Reply:
Thanks for the explanation. I think I got it! Just surprised I didn't tell me there is a cyclic import instead. Would have made things much more obvious :shrug: 

Is there a better place to have this type defined? Or are we ok with having the same thing defined multiple times?
As it was defined twice before I even touched the code.

Reply:
I don't see any big problem in having this defined multiple times. That is just for testing, nothing wild.